### PR TITLE
Make `AMI` optional in Amazon Linux OS detection

### PR DESCRIPTION
**Pull request by jerryaldrichiii**
See:

```
       __|  __|_  )
       _|  (     /   Amazon Linux 2 AMI
      ___|\___|___|

https://aws.amazon.com/amazon-linux-2/
[ec2-user@ip-172-31-47-126 ~]$ cat /etc/system-release
Amazon Linux release 2 (Karoo)
```

More context: https://github.com/inspec/train/pull/312

This closes https://github.com/inspec/train/pull/380 many thanks @artyomtkachenko

### CODE REVIEW

File: lib/train/platforms/detect/helpers/os_linux.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def redhatish_version(conf)
       case conf
       when /rawhide/i
         /((\d+) \(Rawhide\))/i.match(conf)[1].downcase
-      when /Amazon Linux AMI/i
+      when /Amazon Linux (?:AMI|.*)/i
```
Comment (jquick) [inline lib/train/platforms/detect/helpers/os_linux.rb]: Could this just be `/Amazon Linux/i` ?

Comment (jerryaldrichiii) [reply lib/train/platforms/detect/helpers/os_linux.rb]: I'm not sure...I don't see why not though. I'll change it.

**Comment by jerryaldrichiii** [comment]
This will fix https://github.com/inspec/inspec/issues/3614

**Comment by jerryaldrichiii** [comment]
Closing this in favor of #380

**Pull request closed without merging**